skyVegasTests/utilities/validateConsoleMessagesHelper.ts

```diff
@@ -4,17 +4,26 @@ import { messageExists } from '../utilities';
 export const validateConsoleMessages = async (
     expectedMessage: string,
     consoleMessages: string[],
-    intervals: number[] = [5_000, 10_000, 15_000],
-    timeout: number = 60_000
+    deleteMessages = false,
+    intervals: number[] = [5_000, 10_000, 15_000, 20_000],
+    timeout = 60_000
 ) => {
-    if (expectedMessage === undefined) {
-        throw new Error('expected message is null');
-    }
-
-    await expect(async () => {
-        expect(messageExists(consoleMessages, expectedMessage)).toBe(true);
+    await expect(async (): Promise<void> => {
+        const index = messageExists(consoleMessages, expectedMessage);
```

Comment:
I think lines 14 to 17  should be outside the expect assertion declared in line11.  Declare the const index out of the expect and delete the message once the assertion is done. Keeps the expect cleaner.

Reply:
If we move the const index outside the expect it will only be evaluated once as Typescript uses eager evaluation, so once it has been set to false, it will remain as false. If we leave index inside the expect, it will get recalculated again every time we hit a new interval.

Reply:
Violates SRP. Is there no other way to do it ?

Reply:
There is another way to do it, if we use Typescript closures. I have not tried it though. Something like the snippet below

`export default const lazy = function (creator) {
  let res;
  let processed = false;
  return function () {
    if (processed) return res;
    res = creator.apply(this, arguments);
    processed = true;
    return res;
  };
};`

Reply:
Can't really think of a better way to do it apart from checking the console messages twice which will add inefficiencies. I'll leave it like this for now.

Reply:
makespinhelper is the only file that needs to delete messages. It can be invoked after the assertion with consolemessages = []. We then dont need the param deleteMessages passed to the validateconsolemessage function which should only validate console messages.

Reply:
Created 2 separate functions for this now that individually do one thing only (i.e delete or validate)